pkg/internalregistry/controllers/service_account_controller.go

```diff
@@ -165,9 +210,10 @@ func (c *serviceAccountController) sync(ctx context.Context, key string) error {
 	var secretOwnerRefNeedsUpdate, secretSARefNeedsUpdate bool
```

Comment:
Does this also become simpler if the desired apply configuration is built and compared against the existing apply configuration, like you're doing with the ServiceAccount apply, instead of looking for specific fields that are out of level?

Reply:
I do believe that that we can "simplify" in that manner, but I was trying to limit the changes in this PR to the serviceAccount updates.